Content.Client/UserInterface/Systems/Radial/Controls/RadialContainer.xaml.cs

```diff
@@ -0,0 +1,403 @@
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(MoveAnimationTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.Offset),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(new Vector2(0,0), 0f),
+                        new AnimationTrackProperty.KeyFrame(pos, MoveAnimationTime)
+                    }
+                }
+            }
+        };
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    private void PlaySizeAnimation(Control button, float size, string playKey, string? stopKey)
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(FocusAnimationTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.ButtonSize),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(button.Size, 0f),
+                        new AnimationTrackProperty.KeyFrame(new Vector2(size, size), FocusAnimationTime)
+                    }
+                }
+            }
+        };
+
+        if (stopKey != null && button.HasRunningAnimation(stopKey))
+            button.StopAnimation(stopKey);
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        base.Draw(handle);
+
+        foreach (var child in Layout.Children)
+        {
+            var button = (RadialItem)child;
+            LayoutContainer.SetPosition(child, button.Offset - (button.Size/2));
+        }
+
+        // FIXME: We use item's offset like "local item position" for animation. Need make some better way to do it;
+        LayoutContainer.SetPosition(CloseButton, CloseButton.Offset - (CloseButton.Size/2));
+        LayoutContainer.SetPosition(ActionLabel,
+            new Vector2(0 - (GetTextWidth(ActionLabel) / 4), 18.5f + GetDistance(4.5f) ));
+    }
+
+    private int GetTextWidth(RichTextLabel text)
+    {
+        var font = GetFont(text);
+        var msg = text.GetMessage();
+        if (msg == null)
+            return 0;
+
+        var width = 0;
+
+        foreach (var t in msg)
+        {
+            var metrics = font.GetCharMetrics(new Rune(t), 1);
+
+            if (metrics != null)
+                width += metrics.Value.Width + metrics.Value.Advance;
+        }
+
+        return width;
+    }
```

Comment:
This should be on RichTextLabel or using one of the other text dimension methods, not here.